### PR TITLE
add verification that system deletes props for both directions, even …

### PR DESCRIPTION
**Pull request by andriidovhan**
add verification: DELETE /link/props deletes props for both directions, even if only one direction specified

Close: #2388

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy [hunk 1]
```diff
@@ -353,6 +353,10 @@ class LinkSpec extends BaseSpecification {
         assert flow2Path == switchPair.paths.first()
 
         and: "Delete link props from all links of alternative paths to allow rerouting flows"
+        def allLinkProps = northbound.getAllLinkProps()
+        //DELETE /link/props deletes props for both directions, even if only one direction specified
+        northbound.deleteLinkProps([allLinkProps.first()])
+        assert northbound.getAllLinkProps().size() == allLinkProps.size() - 2
         northbound.deleteLinkProps(northbound.getAllLinkProps())
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy]: Why you run the delete operation twice here?
Also, there are plenty of similar places where the same can be verified in LinkPropertiesSpec, why add this strange step in LinkSpec. 
Also, I'm wondering whether the `update` operation is updating both parts of link prop.

Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy]: I lost my train of thoughts(
verification was moved to the `LinkPropertiesSpec.groovy` file

Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy]: > I'm wondering whether the update operation is updating both parts of link prop.

I think so.
if you update one direction then the reversed direction will be updated too

Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy]: > I think so.
should be tested explicitly

Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy]: added

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy [hunk 2]
```diff
@@ -164,6 +164,18 @@ class LinkPropertiesSpec extends BaseSpecification {
         islUtils.getIslInfo(updatedLinks, isl).get().maxBandwidth == maxBandwidthValue.toInteger()
         islUtils.getIslInfo(updatedLinks, isl.reversed).get().maxBandwidth == maxBandwidthValue.toInteger()
 
+        when: "Update cost on forward direction of the link props"
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: Sorry, don't understand the sense of newly added code block. It is already verified at L164 and L165.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy [hunk 3]
```diff
@@ -212,7 +224,11 @@ class LinkPropertiesSpec extends BaseSpecification {
         islUtils.getIslInfo(links, isl.reversed).get().maxBandwidth == maxBandwidthValue.toInteger()
 
         and: "Delete link props"
-        northbound.deleteLinkProps(linkProps)
+        //DELETE /link/props deletes props for both directions, even if only one direction specified
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: And this is verified after L180.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: in mentioned lines we create a new linkProps for an isl
here we update one direction of the created linkProps

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 3 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: unfortunately it is different test

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 2 above)
Comment (ylobankov) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: 1. Ok, got it. So can we improve description at L152 then? For example, `Create link props on the forward direction of ISL to update cost and max bandwidth`.
2. And here `Update link props on the forward direction of ISL to update cost one more time`. 
3. Also, please add some check after L156 that count of created link props is 2.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy [hunk 4]
```diff
@@ -164,6 +164,18 @@ class LinkPropertiesSpec extends BaseSpecification {
         islUtils.getIslInfo(updatedLinks, isl).get().maxBandwidth == maxBandwidthValue.toInteger()
         islUtils.getIslInfo(updatedLinks, isl.reversed).get().maxBandwidth == maxBandwidthValue.toInteger()
 
+        when: "Update cost on forward direction of the link props"
+        def newCostValue = "345"
+        def newLinkPropsInfo = northbound.getAllLinkProps().first().tap { it.props.cost = newCostValue }
+        northbound.updateLinkProps([newLinkPropsInfo])
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: As far as I understand you can do something like that:
`northbound.updateLinkProps([islUtils.toLinkProps(isl, ["cost": newCostValue])])` and remove L169.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy [hunk 5]
```diff
@@ -164,6 +164,18 @@ class LinkPropertiesSpec extends BaseSpecification {
...
+
+        then: "Forward and reverse directions of the link props are really updated"
+        northbound.getAllLinkProps().each {
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: Please, add the same check after L156.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy [hunk 6]
```diff
@@ -164,6 +164,18 @@ class LinkPropertiesSpec extends BaseSpecification {
...
+            assert it.props.cost == newCostValue
+        }
+        database.getIslCost(isl) == newCostValue.toInteger()
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: This check should be in a separate `and` block with description `Cost on forward and reverse ISLs is really updated`

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy [hunk 7]
```diff
@@ -212,7 +224,11 @@ class LinkPropertiesSpec extends BaseSpecification {
         islUtils.getIslInfo(links, isl.reversed).get().maxBandwidth == maxBandwidthValue.toInteger()
 
         and: "Delete link props"
-        northbound.deleteLinkProps(linkProps)
+        //DELETE /link/props deletes props for both directions, even if only one direction specified
+        def allLinkProps = northbound.getAllLinkProps()
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: You don't need to get link props. Use variable `linkProps`. This variable contains one-link direction.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy [hunk 8]
```diff
@@ -212,7 +224,11 @@ class LinkPropertiesSpec extends BaseSpecification {
...
+        allLinkProps.size() == 2
```
Comment (ylobankov) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: Not sure about this check. This check is going to appear in the previous test and it is enough.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: 1. changed to `"Create link props of ISL to update cost and max bandwidth in the forward and reverse directions"`
2. changed to your proposal
3. added

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 5 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: added

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 6 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: new block was added

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 8 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: deleted

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 4 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: updated

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 7 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: thx

**Pull request merged**